lib/internal/quic/core.js

```diff
@@ -2944,14 +2942,14 @@ class QuicStream extends Duplex {
 
     validateObject(headers, 'headers');
 
-    // Push streams are only supported on QUIC servers, and
-    // only if the original stream is bidirectional.
-    // TODO(@jasnell): This is really an http/3 specific
-    // requirement so if we end up later with another
-    // QUIC application protocol that has a similar
-    // notion of push streams without this restriction,
-    // then we'll need to check alpn value here also.
-    if (!this.clientInitiated && !this.bidirectional) {
+    // This is a small performance optimization for http/3,
+    // where push streams are only supported for client
+    // initiated, bidirectional streams. For any other alpn,
+    // we'll make the attempt to push and handle the failure
+    // after.
+    if (!this.clientInitiated &&
+        !this.bidirectional &&
+        this.session.alpnProtocol === 'h3-29') {
```

Comment:
Should this be hardcoded? Will we notice when the protocol changes names again?

Reply:
yeah, we'll definitely notice. The identifier changes with each new draft and each update to ngtcp2. Once it goes final, the `-nn` suffix will be dropped and it'll just be `h3` permanently. Whenever we update ngtcp2, we already have to update this in a couple of other places.